tests/slist_ut/slist_ut.c

```diff
@@ -0,0 +1,1114 @@
+
+    // act
+    PSINGLYLINKEDLIST_ENTRY result = slist_find(&head, test_match_function, TEST_CONTEXT);
+
+    // assert
+    ASSERT_IS_NOT_NULL(result);
+    ASSERT_ARE_EQUAL(void_ptr, head.next, result);
+    ASSERT_ARE_EQUAL(char_ptr, umock_c_get_expected_calls(), umock_c_get_actual_calls());
+}
+
+/* Tests_SRS_SLIST_07_020: [ `slist_find` shall iterate through all entris in the list and return the first entry that satisfies the `match_function` on success. ]*/
+/* Tests_SRS_SLIST_07_021: [ `slist_find` shall determine whether an item satisfies the match criteria by invoking the `match_function` for each entry in the list until a matching entry is found. ]*/
+/* Tests_SRS_SLIST_07_022: [ The `match_function` shall get as arguments the list entry being attempted to be matched and the `match_context` as is. ]*/
+/* Tests_SRS_SLIST_07_024: [ If the `match_function` returns `true`, `slist_find` shall consider that item as matching. ]*/
+/* Tests_SRS_SLIST_07_023: [ If the `match_function` returns `false`, `slist_find` shall consider that item as not matching. ]*/
+TEST_FUNCTION(slist_find_on_a_list_with_2_items_where_the_second_matches_yields_the_second_item)
+{
+    // arrange
+    static simpleItem simp1 = { 1, { NULL } };
+    static simpleItem simp2 = { 2, { NULL } };
+
+    SINGLYLINKEDLIST_ENTRY head;
+    slist_initialize(&head);
+    slist_add(&head, &(simp1.link));
+    slist_add(&head, &(simp2.link));
+
+    STRICT_EXPECTED_CALL(test_match_function(IGNORED_ARG, TEST_CONTEXT))
+        .SetReturn(false);
+    STRICT_EXPECTED_CALL(test_match_function(IGNORED_ARG, TEST_CONTEXT));
+
+    // act
+    PSINGLYLINKEDLIST_ENTRY result = slist_find(&head, test_match_function, TEST_CONTEXT);
+
+    // assert
+    ASSERT_IS_NOT_NULL(result);
+    ASSERT_ARE_EQUAL(void_ptr, &(simp2.link), result);
+    ASSERT_ARE_EQUAL(char_ptr, umock_c_get_expected_calls(), umock_c_get_actual_calls());
+}
+
+/* Tests_SRS_SLIST_07_020: [ `slist_find` shall iterate through all entris in the list and return the first entry that satisfies the `match_function` on success. ]*/
+TEST_FUNCTION(slist_find_on_a_list_with_2_items_both_matching_yields_the_first_item)
+{
+    // arrange
+    static simpleItem simp1 = { 1, { NULL } };
+    static simpleItem simp2 = { 2, { NULL } };
+
+    SINGLYLINKEDLIST_ENTRY head;
+    slist_initialize(&head);
+    slist_add(&head, &(simp1.link));
+    slist_add(&head, &(simp2.link));
+
+    STRICT_EXPECTED_CALL(test_match_function(IGNORED_ARG, TEST_CONTEXT));
+
+    // act
+    PSINGLYLINKEDLIST_ENTRY result = slist_find(&head, test_match_function, TEST_CONTEXT);
+
+    // assert
+    ASSERT_IS_NOT_NULL(result);
+    ASSERT_ARE_EQUAL(void_ptr, head.next, result);
+    ASSERT_ARE_EQUAL(char_ptr, umock_c_get_expected_calls(), umock_c_get_actual_calls());
+}
+
+/* Tests_SRS_SLIST_07_039: [ If the item is not found, `slist_find` shall return `NULL`. ]*/
+TEST_FUNCTION(slist_find_on_a_list_with_2_items_where_none_matches_returns_NULL)
```

Comment:
>slist_find_on_a_list_with_2_items_where_none_matches_returns_NULL [](http://example.com/codeflow?start=14&length=65)

Can we have a test with more than 2 items (like 100)?

Reply:
done